File: FE.Core.FW/src/apis/models/PaymentModel.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type PaymentModel = {
  candidateTopikId?: string | null;
  amount?: number;
  returnUrl?: string | null;
  userName?: string | null;
  profileCode?: string | null;
  koreanName?: string | null;
  vietnameseName?: string | null;
  dob?: string | null;
  examName?: string | null;
  examAreaName?: string | null;
  examLocation?: string | null;
  examAddress?: string | null;
  examDate?: string | null;
  userEmail?: string | null;
  phoneNumber?: string | null;
  examWorkShift?: string | null;
};

export type PaymentSearchModel = {
  candicateName: string | undefined;
  phoneNumber: string | undefined;
  status: string | undefined;
  fromDate: string | undefined;
  toDate: string | undefined;
  transactionNo: string | undefined;
  pageNumber: number;
  pageSize: number;
  examPeriodId: string | undefined;
  userEmail: string | undefined;
};

export type PaymentDetail = {
  paymentRequest: PaymentRequest,
  paymentResponse: PaymentResponse[],
  paymentStatus?: number
};

export type PaymentRequest = {
  id: string,
  txnRef: string,
  amount: string,
  currencyCode: string,
  ipAddress: string,
  createDate: string,
  expireDate: string,
  userName: string | null,
  profileCode: string,
  koreanName: string,
  vietnameseName: string,
  dob: string,
  examName: string,
  examAreaName: string,
  examLocation: string,
  examAddress: string,
  examDate: string,
  userEmail: string,
  phoneNumber: string
}

export type PaymentResponse = {
  id: string,
  responseCode: string,
  responseCodeDescription: string,
  bankCode: string,
  bankTranNo: string,
  cardType: string,
  payDate: string,
  transactionNo: string,
  transactionStatus: string,
  transactionStatusDescription: string,
  responseToVnp: string,
  responseToVnpDescription: string
}

File: BE.Core.FW/Backend/Business/Payment/PaymentModel.cs
using Backend.Infrastructure.EntityFramework.Datatables;

namespace Backend.Business.Payment
{
    public class PaymentModel
    {
        public string CandidateTopikId { get; set; } = null!;
        public ulong Amount { get; set; }
        public string ReturnUrl { get; set; } = null!;
        public string UserName { get; set; } = null!;
        public string ProfileCode { get; set; } = null!;
        public string KoreanName { get; set; } = null!;
        public string VietnameseName { get; set; } = null!;
        public string DOB { get; set; } = null!;
        public string ExamName { get; set; } = null!;
        public string ExamAreaName { get; set; } = null!;
        public string ExamLocation { get; set; } = null!;
        public string ExamAddress { get; set; } = null!;
        public string ExamDate { get; set; } = null!;
        public string UserEmail { get; set; } = null!;
        public string PhoneNumber { get; set; } = null!;
        public string ExamWorkShift { get; set; } = null!;
        public bool UseEnglishLanguage { get; set; } = false;
        public string NoteTimeEnterExamRoom { get; set; } = null!;
    }

    public class ResponseToVnpay
    {
        public string RspCode { get; set; } = null!;
        public string? Message { get; set; }
    }

    public class PaymentHistorySearchModel
    {
        public string? CandicateName { get; set; }
        public string? PhoneNumber { get; set; }
        public int? Status { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public string? TransactionNo { get; set; }
        public Guid? ExamPeriodId { get; set; }
        public string? UserEmail { get; set; }
        public int PageNumber { get; set; } = 1;
        public int PageSize { get; set; } = 20;
    }

    public class PaymentSearchResponse
    {
        public SysPaymentRequestLog PaymentRequest { get; set; } = null!;
        public IEnumerable<SysPaymentResponseLog>? PaymentResponse { get; set; }
        public int? PaymentStatus { get; set; }
    }

    public class PaymentConfirmationEmailModel
    {
        public SysPaymentRequestLog PaymentRequest { get; set; } = null!;
        public SysPaymentResponseLog PaymentResponse { get; set; } = null!;
        public Dictionary<string, string> LocalazationText { get; set; } = new Dictionary<string, string>();
    }
}
